app/ts/components/pages/ConfirmTransaction.tsx

```diff
@@ -169,13 +169,18 @@ export function ConfirmTransaction() {
 			}
 			if (message.method === 'popup_confirm_transaction_dialog_pending_changed') {
 				updatePendingTransactions(message)
-				const currentWindowId = (await browser.windows.getCurrent()).id
-				const currentTabId = (await browser.tabs.getCurrent()).id
-				if (currentWindowId === undefined) throw new Error('could not get current window Id!')
-				if (currentTabId === undefined) throw new Error('could not get current tab Id!')
 				setPendingTransactionAddedNotification(true)
-				browser.windows.update(currentWindowId, { focused: true })
-				browser.tabs.update(currentTabId, { active: true })
+				try {
+					const currentWindowId = (await browser.windows.getCurrent()).id
+					const currentTabId = (await browser.tabs.getCurrent()).id
+					if (currentWindowId === undefined) throw new Error('could not get current window Id!')
```

Comment:
maybe throw early so that it doesn't need to await both tab and window id

```suggestion
					if (currentWindowId === undefined) throw new Error('could not get current window Id!')
					const currentTabId = (await browser.tabs.getCurrent()).id
```